js/main.js

```diff
@@ -9,9 +14,34 @@ xhr.addEventListener('load', loadData);
 xhr.send();
 
 function loadData(event) {
+  data.products = xhr.response;
+  renderData();
+}
+
+$form.addEventListener('submit', handleSubmit);
+
+function handleSubmit(event) {
+  event.preventDefault();
+  var $product = document.querySelectorAll('li');
+  var text = $search.value.toLowerCase();
+  for (var i = 0; i < $product.length; i++) {
+    if (text === $product[i].dataset.producttype) {
+      $list.classList.remove('hidden');
+      $product[i].classList.remove('hidden');
+    } else {
+      $product[i].classList.add('hidden');
+    }
+  }
+  data.view = 'search';
+}
+
+function renderData() {
   for (var i = 0; i < xhr.response.length; i++) {
+    // product list
     var $li = document.createElement('li');
-    $li.className = 'column-third half';
+    $li.className = 'column-third hidden single-product';
+    $li.setAttribute('data-productId', i + 1);
```

Comment:
You should name the attribute `data-product-id`. That way when you use `dataset` to access the value, it will automatically create a property named `productId` for you.